README.md

```diff
@@ -55,6 +55,86 @@ test.all(common)
 
 A valid (read: that follows this interface) IPFS core implementation, must expose the following API.
 
+## Files
+
+### `add`
+
+> Add files and data to IPFS.
+
+##### `Go` **WIP**
+
+##### `JavaScript` - ipfs.files.add(data, [callback])
+
+Where `data` may be
+
+- an array of objects, each of the form
+```js
+{
+  path: '/tmp/myfile.txt',
+  content: (Buffer or Readable stream)
+}
+```
+- a `Buffer` instance
+- a `Readable` stream
+
+`callback` must follow `function (err, res) {}` signature, where `err` is an
+error if the operation was not successful. `res` will be an array of
+[DAGNode][]s.
+
+If no `callback` is passed, a promise is returned.
+
+```js
+var files = [
+  {
+    path: '/tmp/myfile.txt',
+    content: fs.createReadStream('/tmp/myfile.txt')
+  }
+]
+ipfs.files.add(files, function (err, dagNodes) {
+  // 'res' will be an array of DAGNodes
+})
+```
+
+
+### `createAddStream`
+
+> Add files and data to IPFS using a transform stream.
+
+##### `Go` **WIP**
+
+##### `JavaScript` - ipfs.files.createAddStream([callback])
+
+Provides a Transform stream, where objects can be written of the forms
```

Comment:
There are more objects emitted than those who are written. E.g

`/a/b/c.txt`

will emit 3 times, one for the file and 2 other for the dirs.

I understand your reasoning, but feels weird to call it a Transform, it would be like calling a TCP socket a Transform stream, since we will read mutated output. I think both will work

@dignifiedquire @haadcode thoughts?